cpp-linter-lib/src/run.rs

```diff
@@ -130,44 +114,15 @@ pub async fn run_main(args: Vec<String>) -> i32 {
     }
     end_log_group();
 
-    let user_inputs = FeedbackInput {
-        style: args.get_one::<String>("style").unwrap().to_string(),
-        no_lgtm: args.get_flag("no-lgtm"),
-        step_summary: args.get_flag("step-summary"),
-        thread_comments: args
-            .get_one::<String>("thread-comments")
-            .unwrap()
-            .to_string(),
-        file_annotations: args.get_flag("file-annotations"),
-    };
-    let ignore_tidy = args
-        .get_many::<String>("ignore-tidy")
-        .unwrap()
-        .map(|s| s.as_str())
-        .collect::<Vec<_>>();
-    let ignore_format = args
-        .get_many::<String>("ignore-format")
-        .unwrap()
-        .map(|s| s.as_str())
-        .collect::<Vec<_>>();
-
-    let extra_args = convert_extra_arg_val(&args);
-    let mut clang_params = ClangParams {
-        tidy_checks: args.get_one::<String>("tidy-checks").unwrap().to_string(),
-        lines_changed_only,
-        database: database_path,
-        extra_args,
-        database_json: None,
-        style: user_inputs.style.clone(),
-        clang_tidy_command: None,
-        clang_format_command: None,
-        tidy_filter: FileFilter::new(&ignore_tidy, extensions.clone()),
-        format_filter: FileFilter::new(&ignore_format, extensions),
-    };
-    capture_clang_tools_output(&mut arc_files, version, &mut clang_params).await;
+    let mut clang_params = ClangParams::from(&cli);
+    let user_inputs = FeedbackInput::from(&cli);
+    capture_clang_tools_output(&mut arc_files, cli.version.as_str(), &mut clang_params).await;
     start_log_group(String::from("Posting feedback"));
-    rest_api_client.post_feedback(&arc_files, user_inputs).await;
+    let checks_failed = rest_api_client.post_feedback(&arc_files, user_inputs).await;
     end_log_group();
+    if env::var("PRE_COMMIT").is_ok_and(|v| v == "1") {
+        return (checks_failed > 1) as i32;
+    }
```

Comment:
I have to test this externally. `pre-commit` does not support installing hooks from a monorepo (via git source). Furthermore, their rust language support uses very outdated paradigm for build executables from rust code. (see pre-commit/pre-commit#3278)

To test this, I need to start publishing python builds to test-pypi.org, and install the hook from there. So, pre-commit support is on the back burner again...

Reply:
# Reminder
We can install with pip from GitHub source using:
```
pip install git+https://github.com/cpp-linter/cpp_linter_rs.git#subdirectory=cpp-linter-py
```